scipy/stats/_hypotests.py

```diff
@@ -1144,6 +1144,7 @@ def boschloo_exact(table, alternative="two-sided", n=32):
 
     fisher_stat = pvalues[table[0, 0], table[0, 1]]
     index_arr = pvalues <= fisher_stat
+    index_arr += np.isclose(pvalues - fisher_stat, 0)  # Because the hypergeometric has some slight numerical errors
```

Comment:
I would prefer to put this in a separate PR, as it will take some discussion. For instance, currently it appears that this could double-count some `pvalues` that are less than `fisher_stat` and also close to `fisher_stat`.  Also, can you determine a theoretical bound on how close is close enough (i.e. definitely smaller than the smallest legitimate difference between pvalues)?